### PR TITLE
Handle response now returns full response with parsed body

### PR DESCRIPTION
**Pull request by dominic-cicilio**
* **handle_response** now returns the full response (instead of just a parsed body) with an already parsed body.
* All tests pass.

This is a breaking change on all the request methods (get, post, put, delete) because they now return a differently formatted object. Making this a minor release change.

Use case: The ability to have the full response provides access to all of the headers and other http metadata, which can be very useful. Especially the headers.

### CODE REVIEW

File: lib/soda/client.rb [hunk 1]
```diff
@@ -182,22 +182,24 @@ def parse_resource(resource)
...
+      # Return a bunch of mashes as the body if we're JSON
+      begin
+        response.body = JSON.parse(response.body, max_nesting: false)
+        response.body = if response.body.is_a? Array
+                        response.body.map { |r| Hashie::Mash.new(r) }
+                      else
+                        Hashie::Mash.new(response.body)
+                      end
+      rescue
+        raise "JSON parse error"
```
Comment (saraid) [inline lib/soda/client.rb]: Why do this? Just let `JSON.parse` throw a `JSON::ParserError`.

File: .ruby-version [hunk 2]
```diff
@@ -1 +1 @@
-2.2.2
+2.3.2
```
Comment (saraid) [inline .ruby-version]: This seems like an odd decision. We've been on 2.3.6 on a lot of things, and are in the process of moving to 2.5.3.

File: lib/soda/client.rb (hunk 1 above)
Comment (saraid) [reply lib/soda/client.rb]: I... hate this entire gem. Not your fault, but just trying to understand the intention behind it just makes me unhappy.

File: .ruby-version (hunk 2 above)
Comment (dominic-cicilio) [reply .ruby-version]: Oh, I thought this was the project version. If the tests pass with ruby version 2.5.6, should I update this to that version?

**Comment by dominic-cicilio** [comment]
Should the consuming application just call the HTTP requests themselves?

File: .ruby-version (hunk 2 above)
Comment (dominic-cicilio) [reply .ruby-version]: updated to 2.5.3, all tests passed on this version

**Pull request merged**